fl4health/parameter_exchange/parameter_packer.py

```diff
@@ -71,3 +73,47 @@ def unpack_parameters(self, packed_parameters: NDArrays) -> Tuple[NDArrays, List
         model_parameters = packed_parameters[:split_size]
         param_names = packed_parameters[split_size:][0].tolist()
         return model_parameters, param_names
+
+
+class SparseCooParameterPacker(ParameterPacker[Tuple[NDArrays, NDArrays, List[str]]]):
+    """
+    This parameter packer is responsible for selecting an arbitrary set of parameters
+    and then representing them in the sparse COO tensor format, which requires knowing
+    the indices of the parameters within the tensor to which they belong,
+    the shape of that tensor, and also the name of it.
+    """
+
+    def pack_parameters(
+        self, model_parameters: NDArrays, additional_parameters: Tuple[NDArrays, NDArrays, List[str]]
+    ) -> NDArrays:
+        parameter_indices, tensor_shapes, tensor_names = additional_parameters
+        return model_parameters + parameter_indices + tensor_shapes + [np.array(tensor_names)]
+
+    def unpack_parameters(self, packed_parameters: NDArrays) -> Tuple[NDArrays, Tuple[NDArrays, NDArrays, List[str]]]:
+        assert len(packed_parameters) % 3 == 1
```

Comment:
A quick comment on why we expect a modulus of 1 here would be useful, I think

Reply:
I was going to comment this, because it took me a minute to realize the the tensor names is wrapped in a list so its length 1. But I thought maybe I was just slow hah